### PR TITLE
PendingStateManager: Light refactoring/renaming (prep for BatchId)

### PR DESCRIPTION
**Pull request by markfields**
## Description

Some light refactoring:
* Rename `IPendingBatchMessage` to `PendingMessageResubmitData` to better reflect its purpose
* Rename `IBatch.content` to `IBatch.messages`
* Add/use helper for strongly typed access to batch metadata
* Remove redundant `reSubmit` callback for PSM's use; it can/should always use `reSubmitBatch`.
  * Everything is always batches in Turn-Based mode, and there's no harm in treating it the same for Immediate mode.
  * There's no functional change here in this PR, thanks to [this bug fix](https://github.com/markfields/FluidFramework/blob/main/packages/runtime/container-runtime/src/containerRuntime.ts#L1555) a few months back.

## Reviewer Guidance

Suggestion: Hide (leading) whitespace. An if statement is inverted so there's a chunk of indentation-only change in PendingStateManager.

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -661,7 +661,7 @@ export const makeLegacySendBatchFn =
 	(batch: IBatch) => {
 		// Default to negative one to match Container.submitBatch behavior
 		let clientSequenceNumber: number = -1;
-		for (const message of batch.content) {
+		for (const message of batch.messages) {
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/containerRuntime.ts:664]: Is this backwards compatible? Do we need to worry about extremely old batch messages?

Comment (markfields) [reply packages/runtime/container-runtime/src/containerRuntime.ts:664]: If this is a serialized format I will revert it. I assumed it was just a runtime type.  It's not tagged or anything.

Comment (markfields) [reply packages/runtime/container-runtime/src/containerRuntime.ts:664]: Looking at the submit codepath, this type is not directly used for serialization.  Its members get passed around individually over several function calls, final format is `IDocumentMessage`.

**Comment by markfields** [comment]
@tyler-cai-microsoft -- any other concerns before approving?

**Review by tyler-cai-microsoft — APPROVED** [review]

File: packages/runtime/container-runtime/src/opLifecycle/outbox.ts [hunk 2]
```diff
@@ -407,7 +414,10 @@ export class Outbox {
 		return clientSequenceNumber;
 	}
 
-	public checkpoint() {
+	/**
+	 * @returns A checkpoint object per batch that facilitates iterating over the batch messages when rolling back.
+	 */
+	public getBatchCheckpoints() {
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/opLifecycle/outbox.ts:420]: I somehow couldn't find this being used

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -2913,7 +2909,7 @@ export class ContainerRuntime
 			// Note: we are not touching any batches other than mainBatch here, for two reasons:
 			// 1. It would not help, as other batches are flushed independently from main batch.
 			// 2. There is no way to undo process of data store creation, blob creation, ID compressor ops, or other things tracked by other batches.
-			checkpoint = this.outbox.checkpoint().mainBatch;
+			checkpoint = this.outbox.getBatchCheckpoints().mainBatch;
```
Comment (markfields) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2912]: @tyler-cai-microsoft -- it's here

**Pull request merged**